### ISSUE TITLE
[Bug]: Concurrent search failed and worker(10) query failed: [UnexpectedError] Assert "(get_bit(field_data_ready_bitset_, field_id    ))" 

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#23444]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20230412-296380d6
- Deployment mode(standalone or cluster): cluster
- MQ type(rocksmq, pulsar or kafka):  pulsar  
- SDK version(e.g. pymilvus v2.0.0rc2): 2.4.0.dev4
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Search failed after a while od concurrent operations (include insert, delete, search)
```
[2023-04-13 10:00:19,580 - ERROR - fouram]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=10, reason=worker(10) query failed: [UnexpectedError] Assert "(get_bit(field_data_ready_bitset_, field_id))" at /go/src/github.com/milvus-io/milvus/internal/core/src/segcore/SegmentSealedImpl.cpp:643
)>, <Time:{'RPC start': '2023-04-13 10:00:18.432635', 'RPC error': '2023-04-13 10:00:19.580110'}> (decorators.py:108)
[2023-04-13 10:00:19,581 - ERROR - fouram]: Traceback (most recent call last):
  File "/src/fouram/client/util/api_request.py", line 33, in inner_wrapper
    res = func(*args, **kwargs)
  File "/src/fouram/client/util/api_request.py", line 70, in api_request
    return func(*arg, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 660, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 109, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 105, in handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 136, in handler
    ret = func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 85, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 50, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 518, in search
    return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 487, in _execute_search_requests
    raise pre_err
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 478, in _execute_search_requests
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=10, reason=worker(10) query failed: [UnexpectedError] Assert "(get_bit(field_data_ready_bitset_, field_id))" at /go/src/github.com/milvus-io/milvus/internal/core/src/segcore/SegmentSealedImpl.cpp:643
)>
```
 other error:
```
[2023-04-13 10:51:41,839 - ERROR - fouram]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=attempt #0: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_440762547915457080v1 is not available in any replica, err=LackSegment(segmentID=440762547917582236): attempt #1: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_440762547915457080v1 is not available in any replica, err=LackSegment(segmentID=440762547917583128): attempt #2: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_440762547915457080v1 is not available in any replica, err=LackSegment(segmentID=440762547917579954): attempt #3: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_440762547915457080v1 is not available in any replica, err=LackSegment(segmentID=440762547917582747): attempt #4: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_440762547915457080v1 is not available in any replica, err=LackSegment(segmentID=440762547917583585): attempt #5: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_440762547915457080v1 is not available in any replica, err=LackSegment(segmentID=440762547917579991): attempt #6: fail to get shard leaders from QueryCoord: channel by-dev-rootcoord-dml_1_440762547915457080v1 is not available in any replica, err=LackSegment(segmentID=440762547917582244): context done during sleep after run#6: context deadline exceeded)>, <Time:{'RPC start': '2023-04-13 10:51:21.832834', 'RPC error': '2023-04-13 10:51:41.839535'}> (decorators.py:108)
[2023-04-13 10:51:41,839 - ERROR - fouram]: Traceback (most recent call last):
  File "/src/fouram/client/util/api_request.py", line 33, in inner_wrapper
    res = func(*args, **kwargs)
  File "/src/fouram/client/util/api_request.py", line 70, in api_request
    return func(*arg, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 660, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 109, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 105, in handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 136, in handler
    ret = func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 85, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 50, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 518, in search
    return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 487, in _execute_search_requests
    raise pre_err
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 478, in _execute_search_requests
    raise MilvusException(response.status.error_code, response.status.reason)
```

### Steps To Reproduce

```markdown
1. Create collection and create index
2. Insert 1m data and flush
3. Load collection
4. Concurrent: scene_insert_delete_flush and search. After a while, search failed

Case:

        def test_scale_out_querynode(self, input_params: InputParamsBase):
        """
        :test steps:
            1. deploy and insert, load, index, search
            2. upgrade server: scale-out querynode
            3. search test
        """
        # input_params.upgrade_config = {"spec": {"components": {"querynode": {"replicas": 2}}}}
        input_params.upgrade_config = {"queryNode": {"replicas": 2}}

        # before scale: deploy and concurrent search
        input_params.case_skip_clean_collection = True
        default_case_params = ConcurrentParams().params_scene_concurrent(
            [
                ConcurrentParams.params_search(weight=7, nq=1, top_k=10, search_param={"nprobe": 16}, timeout=300),
                ConcurrentParams.params_scene_insert_delete_flush(
                    weight=1, insert_length=10, delete_length=1, random_id=True, random_vector=True, varchar_filled=True)
            ],
            concurrent_number=[20],
            during_time="1h", interval=20,
            **cdp.DefaultIndexParams.IVF_SQ8)
        self.concurrency_template(input_params=input_params, cpu=2, mem=12,
                                  deploy_mode=CLUSTER, old_version_format=False, sync_report=True,
                                  case_callable_obj=ConcurrentClientBase().scene_concurrent_locust,
                                  default_case_params=default_case_params)

        # scale-out query nodes and concurrent test
        input_params.case_skip_prepare = True
        input_params.case_skip_prepare_clean = True

        self.scale_serial_concurrent_template(input_params, deploy_mode=CLUSTER,
                                              case_callable_after_scale=ConcurrentClientBase().scene_concurrent_locust,
                                              default_case_params=default_case_params)
```

### Milvus Log

devops argo name: `fouramf-issue2`

devops cluster, chaos-testing ns pods:
```
fouramf-issue2-93-3122-etcd-0                                    1/1     Running                  0                 61m     10.102.7.249    devops-node11   <none>           <none>
fouramf-issue2-93-3122-etcd-1                                    1/1     Running                  0                 62m     10.102.10.150   devops-node20   <none>           <none>
fouramf-issue2-93-3122-etcd-2                                    1/1     Running                  0                 64m     10.102.5.23     devops-node21   <none>           <none>
fouramf-issue2-93-3122-milvus-datacoord-5dc7df75b9-hdltg         1/1     Running                  1 (130m ago)      134m    10.102.10.81    devops-node20   <none>           <none>
fouramf-issue2-93-3122-milvus-datanode-68b9bd6f79-7xrsn          1/1     Running                  1 (130m ago)      134m    10.102.10.79    devops-node20   <none>           <none>
fouramf-issue2-93-3122-milvus-indexcoord-7575979b6d-m2h5z        1/1     Running                  0                 134m    10.102.10.80    devops-node20   <none>           <none>
fouramf-issue2-93-3122-milvus-indexnode-7dc47df84-qsbhd          1/1     Running                  0                 134m    10.102.7.208    devops-node11   <none>           <none>
fouramf-issue2-93-3122-milvus-proxy-dcc7bcdb-mgtsn               1/1     Running                  1 (130m ago)      134m    10.102.10.84    devops-node20   <none>           <none>
fouramf-issue2-93-3122-milvus-querycoord-5f9476565c-xq579        1/1     Running                  1 (130m ago)      134m    10.102.10.82    devops-node20   <none>           <none>
fouramf-issue2-93-3122-milvus-querynode-74b6f88dcd-5qpvt         1/1     Running                  0                 134m    10.102.7.210    devops-node11   <none>           <none>
fouramf-issue2-93-3122-milvus-querynode-74b6f88dcd-gtjj6         1/1     Running                  0                 64m     10.102.10.149   devops-node20   <none>           <none>
fouramf-issue2-93-3122-milvus-rootcoord-64699d7547-rc7jj         1/1     Running                  1 (130m ago)      134m    10.102.10.83    devops-node20   <none>           <none>
fouramf-issue2-93-3122-minio-0                                   1/1     Running                  0                 134m    10.102.7.230    devops-node11   <none>           <none>
fouramf-issue2-93-3122-minio-1                                   1/1     Running                  0                 134m    10.102.10.86    devops-node20   <none>           <none>
fouramf-issue2-93-3122-minio-2                                   1/1     Running                  0                 134m    10.102.5.234    devops-node21   <none>           <none>
fouramf-issue2-93-3122-minio-3                                   1/1     Running                  0                 134m    10.102.6.126    devops-node10   <none>           <none>
fouramf-issue2-93-3122-pulsar-bookie-0                           1/1     Running                  0                 134m    10.102.7.229    devops-node11   <none>           <none>
fouramf-issue2-93-3122-pulsar-bookie-1                           1/1     Running                  0                 134m    10.102.10.92    devops-node20   <none>           <none>
fouramf-issue2-93-3122-pulsar-bookie-2                           1/1     Running                  0                 134m    10.102.5.238    devops-node21   <none>           <none>
fouramf-issue2-93-3122-pulsar-broker-0                           1/1     Running                  0                 134m    10.102.7.213    devops-node11   <none>           <none>
fouramf-issue2-93-3122-pulsar-proxy-0                            1/1     Running                  0                 134m    10.102.7.212    devops-node11   <none>           <none>
fouramf-issue2-93-3122-pulsar-recovery-0                         1/1     Running                  0                 134m    10.102.7.209    devops-node11   <none>           <none>
fouramf-issue2-93-3122-pulsar-zookeeper-0                        1/1     Running                  0                 134m    10.102.7.228    devops-node11   <none>           <none>
fouramf-issue2-93-3122-pulsar-zookeeper-1                        1/1     Running                  0                 133m    10.102.10.94    devops-node20   <none>           <none>
fouramf-issue2-93-3122-pulsar-zookeeper-2                        1/1     Running                  0                 132m    10.102.5.241    devops-node21   <none>           <none>
```

**Comment by xiaofan-luan** [comment]
that is only on Milvus master but not 2.2 right?

**Comment by ThreadDao** [comment]
> that is only on Milvus master but not 2.2 right?

Yes

**Comment by ThreadDao** [comment]
@weiliu1031
1. argo name in devops: ``
2. server pods in cluster cluster and chaos-testing ns
```
fouramf-issue4-47-1824-etcd-0                                    1/1     Running            0                 62m     10.102.9.90     devops-node13   <none>           <none>
fouramf-issue4-47-1824-etcd-1                                    1/1     Running            0                 63m     10.102.6.233    devops-node10   <none>           <none>
fouramf-issue4-47-1824-etcd-2                                    1/1     Running            0                 64m     10.102.7.74     devops-node11   <none>           <none>
fouramf-issue4-47-1824-milvus-datacoord-846d797775-qvkg2         1/1     Running            0                 135m    10.102.6.53     devops-node10   <none>           <none>
fouramf-issue4-47-1824-milvus-datanode-55c9b9df5c-jcjb9          1/1     Running            1 (131m ago)      135m    10.102.6.52     devops-node10   <none>           <none>
fouramf-issue4-47-1824-milvus-indexcoord-66dc64b57d-7jlcq        1/1     Running            0                 135m    10.102.6.55     devops-node10   <none>           <none>
fouramf-issue4-47-1824-milvus-indexnode-79b775d5c6-fsmxm         1/1     Running            0                 135m    10.102.9.165    devops-node13   <none>           <none>
fouramf-issue4-47-1824-milvus-proxy-6c5d674fc8-z6tbf             1/1     Running            1 (131m ago)      135m    10.102.6.54     devops-node10   <none>           <none>
fouramf-issue4-47-1824-milvus-querycoord-7c897ff4dc-fzm2l        1/1     Running            1 (131m ago)      135m    10.102.6.45     devops-node10   <none>           <none>
fouramf-issue4-47-1824-milvus-querynode-6fb4bcb47d-45f22         1/1     Running            4 (15m ago)       64m     10.102.6.232    devops-node10   <none>           <none>
fouramf-issue4-47-1824-milvus-querynode-6fb4bcb47d-99f7b         0/1     Running            7 (45s ago)       135m    10.102.9.166    devops-node13   <none>           <none>
fouramf-issue4-47-1824-milvus-rootcoord-568d85884c-x2j65         1/1     Running            1 (131m ago)      135m    10.102.6.50     devops-node10   <none>           <none>
fouramf-issue4-47-1824-minio-0                                   1/1     Running            0                 135m    10.102.9.172    devops-node13   <none>           <none>
fouramf-issue4-47-1824-minio-1                                   1/1     Running            0                 135m    10.102.6.76     devops-node10   <none>           <none>
fouramf-issue4-47-1824-minio-2                                   1/1     Running            0                 135m    10.102.7.27     devops-node11   <none>           <none>
fouramf-issue4-47-1824-minio-3                                   1/1     Running            0                 135m    10.102.5.117    devops-node21   <none>           <none>
fouramf-issue4-47-1824-pulsar-bookie-0                           1/1     Running            0                 135m    10.102.6.104    devops-node10   <none>           <none>
fouramf-issue4-47-1824-pulsar-bookie-1                           1/1     Running            0                 135m    10.102.9.174    devops-node13   <none>           <none>
fouramf-issue4-47-1824-pulsar-bookie-2                           1/1     Running            0                 135m    10.102.7.32     devops-node11   <none>           <none>
fouramf-issue4-47-1824-pulsar-broker-0                           1/1     Running            0                 135m    10.102.6.56     devops-node10   <none>           <none>
fouramf-issue4-47-1824-pulsar-proxy-0                            1/1     Running            0                 135m    10.102.9.164    devops-node13   <none>           <none>
fouramf-issue4-47-1824-pulsar-recovery-0                         1/1     Running            0                 135m    10.102.9.163    devops-node13   <none>           <none>
fouramf-issue4-47-1824-pulsar-zookeeper-0                        1/1     Running            0                 135m    10.102.6.82     devops-node10   <none>           <none>
fouramf-issue4-47-1824-pulsar-zookeeper-1                        1/1     Running            0                 134m    10.102.9.176    devops-node13   <none>           <none>
fouramf-issue4-47-1824-pulsar-zookeeper-2                        1/1     Running            0                 134m    10.102.7.34     devops-node11   <none>           <none>
``` 
3. client error:
```
[2023-04-19 13:26:09,391 - ERROR - fouram]: RPC error: [search], <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=9, reason=worker(1) query failed: [UnexpectedError] _Map_base::at)>, <Time:{'RPC start': '2023-04-19 13:26:07.955846', 'RPC error': '2023-04-19 13:26:09.391646'}> (decorators.py:108)
[2023-04-19 13:26:09,394 - ERROR - fouram]: Traceback (most recent call last):
  File "/src/fouram/client/util/api_request.py", line 33, in inner_wrapper
    res = func(*args, **kwargs)
  File "/src/fouram/client/util/api_request.py", line 70, in api_request
    return func(*arg, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/orm/collection.py", line 660, in search
    res = conn.search(self._name, data, anns_field, param, limit, expr,
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 109, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 105, in handler
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 136, in handler
    ret = func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 85, in handler
    raise e
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/decorators.py", line 50, in handler
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 518, in search
    return self._execute_search_requests(requests, timeout, round_decimal=round_decimal, auto_id=auto_id, **kwargs)
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 487, in _execute_search_requests
    raise pre_err
  File "/usr/local/lib/python3.8/dist-packages/pymilvus/client/grpc_handler.py", line 478, in _execute_search_requests
    raise MilvusException(response.status.error_code, response.status.reason)
pymilvus.exceptions.MilvusException: <MilvusException: (code=1, message=fail to search on all shard leaders, err=fail to Search, QueryNode ID=9, reason=worker(1) query failed: [UnexpectedError] _Map_base::at)>
 (api_request.py:48)
```

**Comment by weiliu1031** [comment]
we already ban balance channel operation, please verify this case.

**Comment by ThreadDao** [comment]
Fixed master-20230529-d8ad09b7
argo workflow in devops: fouramf-issue5
